template/utils/uids.py

```diff
@@ -64,7 +64,7 @@ async def get_random_uids(
     avail_uids = []
 
     for uid in range(self.metagraph.n.item()):
-        uid_is_available = await check_uid_availability(
+        uid_is_available = check_uid_availability(
```

Comment:
since `check_uid_availability` is an async function still doesn't this still need to be awaited? And therefore get_random_uids needs to remain async?

Reply:
I think the check_uid_availability needs to be async, and then we just await the check_uid() inside that method